test/e2e/tls_test.go

```diff
@@ -0,0 +1,141 @@
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = Describe("TLS activation", Ordered, func() {
+	const serviceAnnotationKeyTLSSecret = "service.beta.openshift.io/serving-cert-secret-name"
+	var cr *olsv1alpha1.OLSConfig
+	var err error
+	var client *Client
+
+	BeforeAll(func() {
+		client, err = GetClient()
+		Expect(err).NotTo(HaveOccurred())
+		By("Creating a OLSConfig CR")
+		cr, err = generateOLSConfig()
+		Expect(err).NotTo(HaveOccurred())
+		err = client.Create(cr)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterAll(func() {
+		client, err = GetClient()
+		Expect(err).NotTo(HaveOccurred())
+		By("Deleting the OLSConfig CR")
+		Expect(cr).NotTo(BeNil())
+		err = client.Delete(cr)
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	It("should activate TLS on service HTTPS port", func() {
+		By("wait for application server deployment rollout")
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      AppServerDeploymentName,
+				Namespace: OLSNameSpace,
+			},
+		}
+		err = client.WaitForDeploymentRollout(deployment)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Wait for the application service created")
+		service := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      AppServerServiceName,
+				Namespace: OLSNameSpace,
+			},
+		}
+		err = client.WaitForServiceCreated(service)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("check the secret holding TLS ceritificates is created")
+		secretName, ok := service.ObjectMeta.Annotations[serviceAnnotationKeyTLSSecret]
+		Expect(ok).To(BeTrue())
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: OLSNameSpace,
+			},
+		}
+		err = client.WaitForSecretCreated(secret)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("check the deploy has the certificate secret mounted")
+		var secretVolumeDefaultMode = int32(420)
+		Expect(deployment.Spec.Template.Spec.Volumes).To(ContainElement(corev1.Volume{
+			Name: "secret-" + AppServerTLSSecretName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  secretName,
+					DefaultMode: &secretVolumeDefaultMode,
+				},
+			},
+		}))
+
+		By("check HTTPS connection on forwarded port")
+		const inClusterHost = "lightspeed-app-server.openshift-lightspeed.svc.cluster.local"
+		certificate, ok := secret.Data["tls.crt"]
+		Expect(ok).To(BeTrue())
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(certificate)
+
+		var rt http.RoundTripper = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:    caCertPool,
+				ServerName: inClusterHost,
+				// InsecureSkipVerify: true,
```

Comment:
@raptorsun - 1. We can remove this commented line. 2. Do we want to get this client building as separate function to reuse ? 3. We dont have the real /query API test yet , we want a new story ?

Reply:
I will remove this commented line and put this client building procedure into a helper function. 
I can add a real query, too. The lightspeed-service repository will use the operator to deploy a test instance and test with more sophisticated queries. Only 1 test on /query endpoint should be enough for this operator test :)